mobile/src/main/java/org/openhab/habdroid/ui/OpenHABWidgetAdapter.java

```diff
@@ -664,47 +669,87 @@ public void onItemClick(AdapterView<?> parent, View view, int index, long id) {
+                            }
+
+                            final String[] stepValues = new String[(maxValue - minValue)/stepSize +1];
+                            for(int i = 0; i < stepValues.length; i++){
+                                stepValues[i] = String.valueOf(minValue + (i*stepSize));
+                            }
+
+                            AlertDialog.Builder builder = new AlertDialog.Builder(context);
+
+                            if (labelTextView != null) {
+                                builder.setTitle(labelTextView.getText());
+                            }
+                            final LayoutInflater inflater = LayoutInflater.from(context);
+                            final View dialogView = inflater.inflate(R.layout.openhab_dialog_numberpicker, null);
+                            builder.setView(dialogView);
+
+                            // OK button
+                            builder.setPositiveButton(R.string.set, new DialogInterface.OnClickListener() {
+                                public void onClick(DialogInterface dialog, int id) {
+                                    final NumberPicker numberPicker = (NumberPicker) dialogView.findViewById(R.id.numberpicker);
+                                    sendItemCommand(openHABWidget.getItem(), stepValues[numberPicker.getValue()]);
+                                }
+                            });
+                            // Cancel button
+                            builder.setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
+                                public void onClick(DialogInterface dialog, int which) {
+                                    // do nothing, just close the dialog
+                                }
+                            });
+
+                            AlertDialog dialog = builder.create();
+
+                            final NumberPicker numberPicker = (NumberPicker) dialogView.findViewById(R.id.numberpicker);
+
+                            numberPicker.setMinValue(0);
+                            numberPicker.setMaxValue(stepValues.length -1);
+                            numberPicker.setDisplayedValues(stepValues);
+
+                            // Find the closest value in the calculated step values.
+                            int stepIndex = Arrays.binarySearch(stepValues, valueTextView.getText(), new Comparator<CharSequence>() {
+                                @Override
+                                public int compare(CharSequence t1, CharSequence t2) {
+                                    return Integer.valueOf(t1.toString()).compareTo(Integer.valueOf(t2.toString()));
+                                }
+                            });
+                            if ( stepIndex < 0 ){
+                                stepIndex = (-stepIndex) - 1; // Use the returned insertion point if value is not found and select the closest value.
```

Comment:
I have an item set to 10, min 0, max 9 and step 2. The closest value is 8, but 0 is shown instead.